src/cdk/generator.ts

```diff
@@ -131,9 +132,12 @@ export class ResourceGenerator extends Construct {
     // emit CDK constructs for specified resources
     resources.forEach((resource) => this.generateConstructsForResource(resource))
 
+    const namePostfix = randomBetweenNAndM(1000, 2000)
+    const outputName = `ApiBase${namePostfix}`
+
     // it's handy to have the API base URL as a stack output
     if (this.httpApi?.url)
-      new CfnOutput(this, "ApiBase", { value: this.httpApi.url, exportName: Fn.join("-", [Aws.STACK_NAME, "ApiBase"]) })
+      new CfnOutput(this, outputName, { value: this.httpApi.url, exportName: Fn.join("-", [Aws.STACK_NAME, outputName]) })
```

Comment:
I think for the name you could use https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigatewayv2.HttpApi.html#apinamespan-classapi-icon-api-icon-experimental-titlethis-api-element-is-experimental-it-may-change-without-noticespan
```suggestion
      new CfnOutput(this, outputName, { value: this.httpApi.url, exportName: Fn.join("-", [Aws.STACK_NAME, this.httpApi.apiName]) })
```
Might not need the join then either

Reply:
change applied

Reply:
had to use `this.httpApi.apiId` instead  of `this.httpApi.apiName` because I got an error